packages/arb-avm-cpp/avm/src/machine.cpp

```diff
@@ -63,14 +63,18 @@ void MachineExecutionConfig::setSideloadsFromBytes(
     }
 }
 
+void Machine::abort() {
+    is_aborted = true;
+}
+
 Assertion Machine::run() {
     uint256_t start_steps = machine_state.output.total_steps;
     uint256_t start_gas = machine_state.output.arb_gas_used;
 
     bool has_gas_limit = machine_state.context.max_gas != 0;
     BlockReason block_reason = NotBlocked{};
     uint256_t initialConsumed = machine_state.getTotalMessagesRead();
-    while (true) {
+    while (!is_aborted) {
```

Comment:
In addition to only checking this every 100 iterations or so, `is_aborted.load(std::memory_order_relaxed)` should be faster than the default seq_cst load and we don't need any memory order guarantees.

Reply:
:+1: